packages/api/src/utils/stripe.js

```diff
@@ -396,13 +397,196 @@ export function createStripeBillingContext (env) {
     httpClient: Stripe.createFetchHttpClient()
   })
   const billing = StripeBillingService.create(stripe)
+  /** @type {UserCustomerService} */
   const userCustomerService = {
     upsertUserCustomer: env.db.upsertUserCustomer.bind(env.db),
     getUserCustomer: env.db.getUserCustomer.bind(env.db)
   }
   const customers = StripeCustomersService.create(stripe, userCustomerService)
+  const subscriptions = StripeSubscriptionsService.create(stripe)
   return {
     billing,
-    customers
+    customers,
+    subscriptions
   }
 }
+
+/**
+ * @typedef {object} StripeApiForSubscriptionsService
+ * @property {Pick<Stripe['subscriptions'], 'cancel'|'create'>} subscriptions
+ * @property {Pick<Stripe['subscriptionItems'], 'update'|'del'>} subscriptionItems
+ * @property {Pick<Stripe['customers'], 'retrieve'>} customers
+ */
+
+/**
+ * A SubscriptionsService that uses stripe.com for storage
+ */
+export class StripeSubscriptionsService {
+  /**
+   * @param {StripeApiForSubscriptionsService} stripe
+   */
+  static create (stripe) {
+    return new StripeSubscriptionsService(stripe)
+  }
+
+  /**
+   * @param {StripeApiForSubscriptionsService} stripe
+   * @protected
+   */
+  constructor (stripe) {
+    /** @type {StripeApiForSubscriptionsService} */
+    this.stripe = stripe
+    /**
+     * @type {import('./billing-types').SubscriptionsService}
+     */
+    const instance = this // eslint-disable-line
```

Comment:
it's one of the few ways to have tsc type-check that your class `implements` an interface. I found a shorter syntax, though, so will switch to that. Also it may not be necessary at all with the `@returns` in the static `create` func